src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -119,99 +316,1319 @@ const ActorInit En_Egol_InitVars = {
+    element.dim.modelSphere.center.z = centerz
+
+void EnEgol_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    EnEgol* this = THIS;
+
+    this->actor.gravity = -2.0f;
+    Actor_SetScale(&this->actor, 0.015f);
+    this->actor.colChkInfo.damageTable = &sDamageTable;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 60.0f);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.hintId = 0x4A;
+    this->actor.colChkInfo.health = 8;
+    SkelAnime_InitFlex(play, &this->skelAnime, &gEyegoreSkel, &gEyegoreStandAnim, this->jointTable, this->morphTable,
+                       EYEGORE_LIMB_MAX);
+    Collider_InitAndSetJntSph(play, &this->bodyCollider, &this->actor, &sJntSphInit1, this->bodyElements);
+    Collider_InitAndSetJntSph(play, &this->eyeCollider, &this->actor, &sJntSphInit2, this->eyeElements);
+    Collider_InitAndSetQuad(play, &this->laserCollider, &this->actor, &sQuatInit);
+    this->pathIndex = GET_EYEGORE_PATH(&this->actor);
+    if (this->pathIndex == 0x3F) {
+        this->pathIndex = -1;
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    this->wakeupRange = this->actor.world.rot.x * 20.0f;
+    this->actor.world.rot.x = 0;
+    if (this->wakeupRange < 0.0f) {
+        this->wakeupRange = 200.0f;
+    }
+    this->minLaserRange = this->actor.world.rot.z * 20.0f;
+    this->actor.world.rot.z = 0;
+    if (this->minLaserRange < 0.0f) {
+        this->minLaserRange = 200.0f;
+    }
+    this->path = SubS_GetPathByIndex(play, this->pathIndex, 0x3F);
+
+    EYEGORE_SET_DIM(this->eyeCollider.elements[0], 500, 0, 0, 26, 1.0f);
+
+    EYEGORE_SET_DIM(this->bodyCollider.elements[0], 1800, 100, 0, 20, 1.0f);
+    EYEGORE_SET_DIM(this->bodyCollider.elements[1], 1800, 100, 0, 20, 1.0f);
+    EYEGORE_SET_DIM(this->bodyCollider.elements[2], 1000, -500, 0, 35, 1.0f);
+    EYEGORE_SET_DIM(this->bodyCollider.elements[3], 200, 200, 0, 25, 1.0f);
+    EYEGORE_SET_DIM(this->bodyCollider.elements[4], 300, 200, 0, 25, 1.0f);
+    EYEGORE_SET_DIM(this->bodyCollider.elements[5], 2100, -300, 0, 37, 1.0f);
+
+    this->switchFlag = GET_EYEGORE_SWITCH(&this->actor);
+    if (this->switchFlag == 0x7F) {
+        this->switchFlag = -1;
+    }
+    if ((this->switchFlag > -1) && Flags_GetSwitch(play, this->switchFlag)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    EnEgol_SetupWait(this);
+}
+
+void EnEgol_Destroy(Actor* thisx, PlayState* play) {
+    EnEgol* this = THIS;
+
+    Collider_DestroyJntSph(play, &this->bodyCollider);
+    Collider_DestroyJntSph(play, &this->eyeCollider);
+    Collider_DestroyQuad(play, &this->laserCollider);
+}
+
+void EnEgol_SetupWait(EnEgol* this) {
+    this->actor.flags |= ACTOR_FLAG_8000000;
+    this->actionState = EYEGORE_ACTION_WAIT;
+    this->actionFunc = EnEgol_Wait;
+}
+
+void EnEgol_Wait(EnEgol* this, PlayState* play) {
+    f32 frame = this->skelAnime.curFrame;
```

Comment:
I prefer to call this `curFrame`, since that's what it is. idk why enemies like to use this pattern.
```suggestion
    f32 curFrame = this->skelAnime.curFrame;
```